pkg/build/controller/strategy/util.go

```diff
@@ -649,3 +653,106 @@ func setupBlobCache(pod *corev1.Pod) {
 		pod.Spec.InitContainers = initContainers
 	}
 }
+
+// injectBuildVolumesIntoPod injects the provided BuildVolumes into given Pod
+func addBuildVolumesToPod(pod *corev1.Pod, buildVolumes []buildv1.BuildVolume) error {
+	if len(buildVolumes) == 0 {
+		return nil
+	}
+
+	for _, bv := range buildVolumes {
+		v := corev1.Volume{
+			Name: bv.Name,
+		}
+
+		switch {
+		case bv.Type == buildv1.BuildVolumeSourceTypeSecret:
+			v.Secret = bv.Secret
+		case bv.Type == buildv1.BuildVolumeSourceTypeConfigMap:
+			v.ConfigMap = bv.ConfigMap
+		default:
+			return fmt.Errorf("encountered unsupported build volume source type %q", bv.Type)
+		}
+
+		pod.Spec.Volumes = append(pod.Spec.Volumes, v)
+	}
+
+	return nil
+}
+
+// injectBuildVolumeMountsIntoContainer injects the provided BuildVolumeMounts into the given Container
+func addBuildVolumeMountsToContainer(container *corev1.Container, buildVolumeMounts []buildv1.BuildVolumeMount) error {
+	if len(buildVolumeMounts) == 0 {
+		return nil
+	}
+
+	for _, bvm := range buildVolumeMounts {
+		vm := corev1.VolumeMount{
+			Name:      bvm.Name,
+			ReadOnly:  bvm.ReadOnly,
+			MountPath: path.Join(BuildVolumeMountPath, bvm.Name),
```

Comment:
@coreydaley Now I see what you are doing in https://github.com/openshift/builder/pull/245#discussion_r632530761.

These need to be consistent - ideally belonging in library-go (we already have a buildutil package there).

Reply:
moved to library-go